<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Riley Xin" />


<title>Dealing with Multiple Mutations</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Dealing with Multiple Mutations</h1>
<h4 class="author">Riley Xin</h4>



<div id="overview" class="section level2">
<h2>Overview</h2>
<p>In the introduction, we consider an experiment primarily focusing on
single amino acid substitutions, with each variant being a point
mutation. <code>Rosace</code> can also be used to analyze data with
variants containing multiple mutations at various positions. You can
either (1) run Rosace without positional information or (2) provide a
positional grouping to allow information to be shared across similar
variants. Since the number of variants with multiple mutations at the
exact same positions might be small, (2) requires using heuristics to
group variants at similar positions together. We demonstrate two
clustering methods using the <a href="https://mavedb.org/#/score-sets/urn:mavedb:00000003-a-1/">BRCA1
RING</a> phage display data provided by <code>Enrich2</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;rosace&quot;</span>)</span></code></pre></div>
</div>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<p>We’ve processed the BRCA1 RING data to include nucleotide-level and
amino acid-level mutation positions, as well as wildtypes, mutations,
and mutation types. The amino acid-level positional label will be used
for grouping.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;BRCA1R&quot;</span>)</span></code></pre></div>
</div>
<div id="create-rosace-object" class="section level2">
<h2>Create Rosace object</h2>
<p>Next, create a Rosace object and run data imputation, normalization,
and normalization. The BRCA1 phage experiment contains 6 replicates.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>key <span class="ot">&lt;-</span> <span class="st">&quot;BRCA1R&quot;</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>type <span class="ot">&lt;-</span> <span class="st">&quot;growth&quot;</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>assay1 <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(BRCA1R.data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">18</span><span class="sc">:</span><span class="dv">23</span>)),</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>                            <span class="at">var.names =</span> BRCA1R.data<span class="sc">$</span>hgvs_nt,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>                            <span class="at">key =</span> key, <span class="at">rep =</span> <span class="dv">1</span>, <span class="at">type =</span> type)</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>assay2 <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(BRCA1R.data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">24</span><span class="sc">:</span><span class="dv">29</span>)),</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>                            <span class="at">var.names =</span> BRCA1R.data<span class="sc">$</span>hgvs_nt,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>                            <span class="at">key =</span> key, <span class="at">rep =</span> <span class="dv">2</span>, <span class="at">type =</span> type)</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>assay3 <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(BRCA1R.data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">30</span><span class="sc">:</span><span class="dv">35</span>)),</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>                            <span class="at">var.names =</span> BRCA1R.data<span class="sc">$</span>hgvs_nt,</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>                            <span class="at">key =</span> key, <span class="at">rep =</span> <span class="dv">3</span>, <span class="at">type =</span> type)</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>assay4 <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(BRCA1R.data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">36</span><span class="sc">:</span><span class="dv">41</span>)),</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>                            <span class="at">var.names =</span> BRCA1R.data<span class="sc">$</span>hgvs_nt,</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>                            <span class="at">key =</span> key, <span class="at">rep =</span> <span class="dv">4</span>, <span class="at">type =</span> type)</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>assay5 <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(BRCA1R.data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">42</span><span class="sc">:</span><span class="dv">47</span>)),</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>                            <span class="at">var.names =</span> BRCA1R.data<span class="sc">$</span>hgvs_nt,</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>                            <span class="at">key =</span> key, <span class="at">rep =</span> <span class="dv">5</span>, <span class="at">type =</span> type)</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>assay6 <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(BRCA1R.data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">48</span><span class="sc">:</span><span class="dv">53</span>)),</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>                            <span class="at">var.names =</span> BRCA1R.data<span class="sc">$</span>hgvs_nt,</span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>                            <span class="at">key =</span> key, <span class="at">rep =</span> <span class="dv">6</span>, <span class="at">type =</span> type)</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">CreateRosaceObject</span>(<span class="at">object =</span> assay1)</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">AddAssayData</span>(<span class="at">object =</span> rosace, <span class="at">assay =</span> assay2)</span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">AddAssayData</span>(<span class="at">object =</span> rosace, <span class="at">assay =</span> assay3)</span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">AddAssayData</span>(<span class="at">object =</span> rosace, <span class="at">assay =</span> assay4)</span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">AddAssayData</span>(<span class="at">object =</span> rosace, <span class="at">assay =</span> assay5)</span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">AddAssayData</span>(<span class="at">object =</span> rosace, <span class="at">assay =</span> assay6)</span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="fu">GetAssayName</span>(rosace)</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">ImputeData</span>(rosace, <span class="at">key =</span> key, <span class="at">impute.method =</span> <span class="st">&quot;zero&quot;</span>)</span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(rosace, <span class="at">key =</span> key,</span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a>                        <span class="at">normalization.method =</span> <span class="st">&quot;wt&quot;</span>,</span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a>                        <span class="at">wt.var.names =</span> <span class="st">&quot;_wt&quot;</span>,</span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>                        <span class="at">wt.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">IntegrateData</span>(<span class="at">object =</span> rosace, <span class="at">key =</span> key)</span></code></pre></div>
</div>
<div id="run-rosace-without-positional-information" class="section level2">
<h2>Run Rosace without positional information</h2>
<p>Add variant data including hgvs name, aa_pos_mut, wildtype, mutation,
and mutation type to the Rosace object. Running Rosace without
specifying the ‘pos.col’ argument will compute variant scores without
leveraging positional information. For demonstration purposes, only the
first 100 variants are used.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>rosace<span class="sc">@</span>var.data <span class="ot">&lt;-</span> rosace<span class="sc">@</span>var.data <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(BRCA1R.data[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;variants&quot;</span> <span class="ot">=</span> <span class="st">&quot;hgvs_nt&quot;</span>))</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">head</span>(rosace<span class="sc">@</span>var.data)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>raw.counts <span class="ot">&lt;-</span> rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>raw.counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ]</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>combined.counts <span class="ot">&lt;-</span> rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>combined.counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ]</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>var.names <span class="ot">&lt;-</span> rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>var.names[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">RunRosace</span>(<span class="at">object =</span> rosace,</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;BRCA1R&quot;</span>,</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>                    <span class="at">type =</span> <span class="st">&quot;AssaySet&quot;</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>                    <span class="at">savedir =</span> <span class="st">&quot;../tests/results/stan/assayset/&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>                    <span class="at">ctrl.col =</span> <span class="st">&quot;type&quot;</span>,</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>                    <span class="at">ctrl.name =</span> <span class="st">&quot;synonymous&quot;</span>,</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>                    <span class="at">install =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="run-rosace-with-positional-information" class="section level2">
<h2>Run Rosace with positional information</h2>
<p>Running Rosace with positional information requires additional
grouping when there are not enough variants at the exact same positions
(less than 10). One option is to use unsupurvised clustering on the
positions. First, create a position matrix for variants with multiple
mutations:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>pos.mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">lapply</span>(<span class="fu">strsplit</span>(BRCA1R.data<span class="sc">$</span>aa_pos_mut, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>), as.numeric), <span class="cf">function</span>(x){</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>   <span class="fu">c</span>(x,<span class="fu">rep</span>(<span class="dv">0</span>,<span class="dv">10</span><span class="sc">-</span><span class="fu">length</span>(x)))</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>})), <span class="at">ncol =</span> <span class="dv">10</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="fu">rownames</span>(pos.mat) <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">dim</span>(BRCA1R.data)[<span class="dv">1</span>])</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>pos.mat <span class="ot">&lt;-</span> pos.mat[pos.mat[, <span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">0</span>,]</span></code></pre></div>
<p>Then, consider clustering algorithms such as <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/kmeans">kmeans</a>
or <a href="https://cran.r-project.org/web/packages/scclust/readme/README.html">scclust</a>.
Kmeans requires specifying the number of groups, while
<code>scclust</code> allows one to specify the minimum number of
elements in a group.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># group variants using kmeans</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>clust_kmeans <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(pos.mat, <span class="at">centers =</span> <span class="dv">500</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>clust.res <span class="ot">&lt;-</span> clust_kmeans<span class="sc">$</span>cluster</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># group variants using scclust</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;scclust&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;scclust&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>dist <span class="ot">&lt;-</span> <span class="fu">distances</span>(<span class="fu">data.frame</span>(pos.mat, <span class="at">idx =</span> <span class="fu">rownames</span>(pos.mat)),</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                  <span class="at">id_variable =</span> <span class="st">&quot;idx&quot;</span>,</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>                  <span class="at">dist_variables =</span> <span class="fu">paste0</span>(<span class="st">&quot;X&quot;</span>, <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">10</span>)))</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co"># optimizing the clusters while ensuring a minimum of 10 variants in each cluster </span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>clust.res <span class="ot">&lt;-</span> <span class="fu">sc_clustering</span>(dist, <span class="dv">10</span>)</span></code></pre></div>
<p>Label position groups for both single mutations (from original
position) and multiple mutations (from clustering assignment) in a
column ‘pos_group’.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>BRCA1R.data <span class="ot">&lt;-</span> BRCA1R.data <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pos_group =</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>,aa_pos_mut), <span class="at">.after =</span> <span class="dv">4</span>)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>BRCA1R.data[<span class="fu">rownames</span>(pos.mat),]<span class="sc">$</span>pos_group <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M&quot;</span>, clust.res)</span></code></pre></div>
<p>Then, run Rosace with the ‘pos.col’ argument being ‘pos_group’:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>rosace<span class="sc">@</span>var.data <span class="ot">&lt;-</span> rosace<span class="sc">@</span>var.data <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(BRCA1R.data[, <span class="dv">2</span><span class="sc">:</span><span class="dv">8</span>], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;variants&quot;</span> <span class="ot">=</span> <span class="st">&quot;hgvs_nt&quot;</span>))</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">head</span>(rosace<span class="sc">@</span>var.data)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>raw.counts <span class="ot">&lt;-</span> rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>raw.counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ]</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>combined.counts <span class="ot">&lt;-</span> rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>combined.counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, ]</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>var.names <span class="ot">&lt;-</span> rosace<span class="sc">@</span>assay.sets<span class="sc">$</span><span class="st">`</span><span class="at">BRCA1R</span><span class="st">`</span><span class="sc">@</span>var.names[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>rosace <span class="ot">&lt;-</span> <span class="fu">RunRosace</span>(<span class="at">object =</span> rosace,</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;BRCA1R&quot;</span>,</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>                    <span class="at">type =</span> <span class="st">&quot;AssaySet&quot;</span>,</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>                    <span class="at">savedir =</span> <span class="st">&quot;../tests/results/stan/assayset/&quot;</span>,</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>                    <span class="at">pos.col =</span> <span class="st">&quot;pos_group&quot;</span>,</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>                    <span class="at">ctrl.col =</span> <span class="st">&quot;type&quot;</span>,</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>                    <span class="at">ctrl.name =</span> <span class="st">&quot;synonymous&quot;</span>,</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>                    <span class="at">install =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Finally, get the variant scores and proceed with downstream
analysis.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>scores.data.list<span class="ot">&lt;-</span> <span class="fu">OutputScore</span>(rosace, <span class="at">pos.info =</span> <span class="cn">TRUE</span>, <span class="at">name =</span> <span class="st">&quot;BRCA1R_ROSACE1&quot;</span>, <span class="at">sig.test =</span> <span class="fl">0.05</span>)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>scores.pos <span class="ot">&lt;-</span> scores.data.list<span class="sc">$</span>df_position</span></code></pre></div>
<div style="height: 200px;">

</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
